python/lsst/analysis/drp/qaUtils.py

```diff
@@ -0,0 +1,225 @@
+             "HSC-I": -0.461, "HSC-Z": 0.0, "const": -0.478}
+    xPara = {"name": "xPara", "range": "griRed", "HSC-G": 0.075, "HSC-R": 0.922,
+             "HSC-I": -0.997, "HSC-Z": 0.0, "const": -1.442}
+    yPara = {"name": "yPara", "range": "rizRed", "HSC-G": 0.0, "HSC-R": 0.928,
+             "HSC-I": -0.557, "HSC-Z": -0.372, "const": -1.332}
+
+    # Straight line fits for the perpendicular ranges
+    # The following fits were derived in the process of calibrating
+    # the above coeffs (all three RC2 tracts gave ~ the same fits).
+    # May remove later if deemed no longer useful.
+    wFit = {"name": "wFit", "range": "griBlue", "HSC-G": 0.52, "HSC-R": -0.52,
+            "HSC-I": 0.0, "HSC-Z": 0.0, "const": -0.08}
+    xFit = {"name": "xFit", "range": "griRed", "HSC-G": 13.35, "HSC-R": -13.35,
+            "HSC-I": 0.0, "HSC-Z": 0.0, "const": -15.54}
+    yFit = {"name": "yFit", "range": "rizRed", "HSC-G": 0.0, "HSC-R": 0.40,
+            "HSC-I": -0.40, "HSC-Z": 0.0, "const": 0.03}
+
+    # Use the equations from Ivezic 2004 to calculate these values for each
+    # source
+    for coeffs in [wPerp, xPerp, yPerp, wPara, xPara, yPara, wFit, xFit, yFit]:
+        vals = (cat["g" + magColName].values*coeffs["HSC-G"]
+                + cat["r" + magColName].values*coeffs["HSC-R"]
+                + cat["i" + magColName].values*coeffs["HSC-I"]
+                + cat["z" + magColName].values*coeffs["HSC-Z"] + coeffs["const"])
+        cat[coeffs["name"]] = vals
+
+    return cat
+
+
+def addGalacticExtinction(cat):
+    """Adds galactic extinction for each source
+
+    Parameters
+    ----------
+    cat : `pandas.core.frame.DataFrame`
+
+    Returns
+    -------
+    cat : `pandas.core.frame.DataFrame`
+    """
+    pass
+
+
+def addUseForQAFlag(cat):
+    """Add a flag to say if the source should be used for QA purposes
+
+    Parameters
+    ----------
+    cat : `pandas.core.frame.DataFrame`
+
+    Returns
+    -------
+    cat : `pandas.core.frame.DataFrame`
+    """
+
+    # TODO: This needs to be checked and probably changed
+
+    use = ((np.isfinite(cat["gPsMag"])) & (np.isfinite(cat["rPsMag"]))
+           & (np.isfinite(cat["iPsMag"])) & (np.isfinite(cat["zPsMag"]))
+           & (np.isfinite(cat["zPsMag"])) & (np.isfinite(cat["gCModelMag"]))
+           & (np.isfinite(cat["rCModelMag"])) & (np.isfinite(cat["iCModelMag"]))
+           & (np.isfinite(cat["zCModelMag"])) & (np.isfinite(cat["yCModelMag"])))
+
+    for band in ["g", "r", "i", "z", "y"]:
+        usePerBand = ((cat[band + "Centroid_flag_notAtMaximum"] == 0) & (cat[band + "Shape_flag"] == 0)
+                      & (cat[band + "PsfFlux_flag"] == 0)
+                      & (cat[band + "PixelFlags_saturatedCenter"] == 0)
+                      & (cat[band + "Extendedness_flag"] == 0)
+                      & (cat[band + "Deblend_nChild"] == 0))
+        use = use & usePerBand
+    cat["useForQAFlag"] = use
+    return cat
```

Comment:
What about something like this?
```
use = np.ones(len(cat)).astype(bool)

for b in bands:  # add bands as a keyword arg above
    for magType in magTypes:  # add magTypes as a kwarg too
        use &= np.isfinite(cat[b + magType + 'Mag'])
    for flag in flagList:  # make flagList a kwarg
        use &= cat[band + flag] == 0

    cat.loc[:, 'useForQAFlag'] = use
    return cat
```

Reply:
keeping with the general pattern to make all these more customizable if needed ever...  which makes me wonder...why aren't all these functions actions?  Seems like a general action could`appendAllTheColumns`.  Probably not necessary, but just wondering why the actions stop here.